internal/login/login.go

```diff
@@ -35,6 +35,9 @@ import (
 	"github.com/docker/hub-tool/internal/hub"
 )
 
+// ErrCanceled represents a normally canceled operation
+var ErrCanceled = errors.New("canceled")
```

Comment:
This error is present 3 times in the change, we could create a package `errdefs` and reuse it?

Reply:
I also thought about it. and eventually decided to put it under the command's package because I assume we may want to distinguish between login cancel error and other commands.
IMHO, one package/file that contains is less elegant for this use-case...

However, it's a minor change - let me know what do you prefer and I will set it up

Reply:
My thinking was: if we have one error for the case when the user cancels the current action then we can [in main.go](https://github.com/docker/hub-tool/blob/main/main.go#L73) handle that error and return the right exit code (130).

Reply:
I got you, missed the exit code part, in this case your suggestion sounds pretty good 👍 

There is a tricky part here, if we will return an error from the cobra handler the error will be printed (cobra - command.go:941) unless we will turn off the `SilenceErrors` which I don't think it's a good idea because other errors won't be displayed.

Do you have any other idea besides making the changes in the cobra library?

Reply:
Ah right! The errors, indeed :)

In compose-cli we do [silence the errors](https://github.com/docker/compose-cli/blob/main/cli/main.go#L106) but then we [print them manually](https://github.com/docker/compose-cli/blob/main/cli/main.go#L245).

Reply:
Anyway, I made the changes you suggested with the errors declaration.
For the silence-errors stuff, I think it would be better to be discussed in a separate issue considering if the exit-code handling is worth those changes.

Reply:
Thanks! @silvin-lubecki will review this tomorrow, and hopefully it will be merged then :)

Thanks!